js-miniapp-sample/src/pages/universal-bridge.js

```diff
@@ -0,0 +1,134 @@
+    paddingTop: 20,
+    paddingBottom: 20,
+  },
+  card: {
+    width: '100%',
+    height: 'auto',
+  },
+  actions: {
+    justifyContent: 'center',
+  },
+  content: {
+    justifyContent: 'center',
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    fontSize: 18,
+    color: theme.color.primary,
+    fontWeight: 'bold',
+    paddingBottom: 0,
+  },
+  textfield: {
+    width: '90%',
+    maxWidth: 300,
+    marginTop: 10,
+    marginBottom: 10,
+    background: 'white',
+    '& input': {
+      color: theme.color.primary,
+      lineHeight: '1.5em',
+      fontSize: '1.2em',
+    },
+  },
+}));
+
+const UniversalBridge = () => {
+  const classes = useStyles();
+  const defaultJsonValue = '{"data":"This is a sample json information"}';
+  let inputValue = defaultJsonValue;
+  let [receiveJsonInfo, setReceiveJsonInfo] = useState('');
+
+  window.addEventListener(HostAppEvents.RECEIVE_JSON_INFO, function (e) {
+    let message = e.detail.message;
+    console.log(message);
+    receiveJsonInfo = message;
+    setReceiveJsonInfo(message);
+  });
+
+  const handleInput = (e: SyntheticInputEvent<HTMLInputElement>) => {
+    e.preventDefault();
+    inputValue = e.currentTarget.value;
+  };
+
+  const sendJson = () => {
+    const info = { content: inputValue };
+    MiniApp.universalBridge
+      .sendJsonToHostapp(info)
+      .then((success) => {
+        console.log(success);
+      })
+      .catch((miniAppError) => {
+        console.error('Send Json Error: ', miniAppError);
+      });
+  };
+
+  return (
+    <div className={classes.scrollable}>
+      <GreyCard className={classes.card}>
+        <CardContent className={classes.content}>
+          <p>Send Json to HostApp</p>
+        </CardContent>
+        <CardContent className={classes.content}>
+          <TextField
+            type="text"
+            className={classes.textfield}
+            onChange={handleInput}
+            placeholder="Content"
+            defaultValue={defaultJsonValue}
+            variant="outlined"
+            color="primary"
+            multiline="true"
+            inputProps={{
+              'data-testid': 'input-field',
+            }}
+          />
+        </CardContent>
+        <CardActions className={classes.actions}>
+          <Button
+            color="primary"
+            className={classes.button}
+            onClick={sendJson}
+            variant="contained"
+          >
+            Send Json
+          </Button>
+        </CardActions>
+        <hr />
+        <CardContent className={classes.content}>
+          <p>Received Json from HostApp</p>
+        </CardContent>
+        <CardContent className={classes.content}>
+          <TextField
+            type="text"
+            className={classes.textfield}
+            placeholder="Received Json info here..."
+            value={receiveJsonInfo}
+            variant="outlined"
+            color="primary"
+            multiline="true"
+          />
```

Comment:
We can use a lable or disable editing for this textfield. Because mini-app would just receive the info from the Hostapp.

Reply:
Yeah, I have disabled the textfield.